### PR TITLE
func: Allow custom paths to be added the the getter landlock

### PR DESCRIPTION
**Pull request by astudentofblake**
Fixes: #20315

### CODE REVIEW

**Review by shoenig — CHANGES_REQUESTED** [review]
This is excellent @astudentofblake! Just had a few minor nitpicks to clear up and then we can get this merged

File: client/config/artifact.go [hunk 1]
```diff
@@ -9,6 +9,7 @@ import (
 
 	"github.com/dustin/go-humanize"
 	"github.com/hashicorp/nomad/nomad/structs/config"
+	"golang.org/x/exp/slices"
```
[the remark this answers is not in the document]
Comment (astudentofblake) [reply client/config/artifact.go]: I've fixed these.

"golang.org/x/exp/slices" still exists in 2 other places outside this review, there seems to be a semgrep rule to catch these, but I didn't notice an error

File: nomad/structs/config/artifact.go [hunk 2]
```diff
@@ -9,7 +9,10 @@ import (
 	"time"
 
 	"github.com/dustin/go-humanize"
+	"github.com/hashicorp/nomad/helper"
 	"github.com/hashicorp/nomad/helper/pointer"
+	"github.com/shoenig/go-landlock"
+	"golang.org/x/exp/slices"
```
[the remark this answers is not in the document]
Comment (astudentofblake) [reply nomad/structs/config/artifact.go]: Fixed

File: nomad/structs/config/artifact_test.go [hunk 3]
```diff
@@ -15,6 +15,11 @@ func TestArtifactConfig_Copy(t *testing.T) {
 	ci.Parallel(t)
 
 	a := DefaultArtifactConfig()
+	a.FilesystemIsolationExtraPaths = []string{
+		"p:a/b:r",
+		"p:a/c/drw",
+		"f:x/y:rw",
```
[the remark this answers is not in the document]
Comment (astudentofblake) [reply nomad/structs/config/artifact_test.go]: These should all now contain reasonable data

File: website/content/docs/configuration/client.mdx [hunk 4]
```diff
@@ -436,6 +436,10 @@ see the [drivers documentation](/nomad/docs/drivers).
   isolation should be disabled for artifact downloads. Applies only to systems
   where filesystem isolation via [landlock] is possible (Linux kernel 5.13+).
 
+- `filesystem_isolation_extra_paths` `([]string: nil)` - Allow extra paths
+  in the filesystem isolation. Paths are specified in the form "[kind]:[mode]:[path]"
+  e.g. "f:r:/dev/urandom"
```
[the remark this answers is not in the document]
Comment (astudentofblake) [reply website/content/docs/configuration/client.mdx]: I've updated this with the extra info and tried to match the formatting.

File: nomad/structs/config/artifact.go [hunk 5]
```diff
@@ -209,6 +227,12 @@ func (a *ArtifactConfig) Validate() error {
 		return fmt.Errorf("disable_filesystem_isolation must be set")
 	}
 
+	for _, p := range a.FilesystemIsolationExtraPaths {
+		if _, err := landlock.ParsePath(p); err != nil {
+			return fmt.Errorf("filesystem_isolation_extra_paths contains invalid lockdown path %s", p)
```
[the remark this answers is not in the document]
Comment (astudentofblake) [reply nomad/structs/config/artifact.go]: fixed

**Review by shoenig — APPROVED** [review]
LGTM, thanks @astudentofblake!

**Pull request merged**